### DISCUSSION TITLE
Trying to upload a file using Formik but get an InvalidStateError: Failed to set the 'value' property on 'HTMLInputElement':

### DISCUSSION THREAD

**Discussion by Stephane-Pires** [jaredpalmer/formik#3128] [Q&A]
I'm trying to upload a simple image file.

But can't understand why i'm getting

`InvalidStateError: Failed to set the 'value' property on 'HTMLInputElement': This input element accepts a filename, which may only be programmatically set to the empty string.`

Here's my code :) 

```jsx

export const ImageUploaderField = ({ ...props }) => {
    const { setFieldValue } = useFormikContext()
    const [field] = useField(props)

    return (
        <Field
            {...field}
            {...props}
            onChange={(event) => {
                setFieldValue(field.name, event.currentTarget.files[0])
            }}
        />
    )
}
```

Usage

```jsx
<ImageUploaderField name="filename" type="file" />

```

I have been struggling with this the whole day. 

Also, i'm new to Formik, but i know a bit about React :)

**Comment by josangelATM** [comment]
Same here :(

**Reply by josangelATM** [reply]
Guys if you need, is a weird solution but its working for me.


https://stackoverflow.com/questions/66876022/setfieldvalue-formik-and-invalidstateerror-failed-to-set-the-value-property

**Reply by voranzovv** [reply]
@josangelATM Thanks buddy

**Reply by evchip** [reply]
This worked, thanks!

**Reply by ghost** [reply]
worked!!

**Reply by ForJessicaSake** [reply]
you saved my life lol, thanks

**Comment by chinanderm** [comment]
The value of `input` with `type` "file" can only be set by the user, not code. Thus, supplying a `value` attribute/prop results in the error you're getting. Instead of using `<Field />` you should instead add the relevant `onChange`, `onBlur`, etc fields yourself.

Here's how I have approached it:
```
<MyFileInput
  inputProps={{
    id: 'logoFile',
    name: 'logoFile',
    onBlur: () => setFieldTouched('myFile'),
    onChange: e => setFieldValue('myFile', e.target.files),
  }}
  label="Select a File"
  onRemove={() => {
    setFieldValue('myFile', null)
  }}
/>

... in MyFileInput ...
<input type="file" {...props.inputProps} />
```

**Reply by BlackBeardo** [reply]
I had same error in my custom react-dropzone component. i removed `value` and now it's working. I have some unhealthy(atleast I think it is) for file uploader component with dropzone + controlled input from react-form-hook + Yup. 

[I found it here](https://itecnote.com/tecnote/angular-failed-to-set-the-value-property-on-htmlinputelement/).
its for angular, but core logic was same for my react component.

if you want I can show some code snippets, if that helps. let me know. not pasting for now.

**Comment by anhtuanleee** [comment]
i was same problem with react-hook-form, u should set value of input to undefined, will fixed, i think checking of form incorrect

![image](https://github.com/user-attachments/assets/b24cfcc8-36e2-41fd-9722-c498bc963c9f)

**Reply by LipanRyNeX** [reply]
Thanks for the fix man , i wasted so much time with this problem.

**Reply by anhtuanleee** [reply]
> Thanks for the fix man , i wasted so much time with this problem.

ur wellcome

**Reply by tw4** [reply]
this is great dude thx

**Reply by A1X5H04** [reply]
Hey, which font you are using ?

**Reply by anhtuanleee** [reply]
> Hey, which font you are using ?

 i use jetbraindmono bro
